cnd/src/btsieve/block_by_hash.rs

```diff
@@ -0,0 +1,44 @@
+macro_rules! impl_block_by_hash {
+    () => {
+        impl<C> BlockByHash for Cache<C>
```

Comment:
Macros should never rely on their context to import certain things.

I would suggest to make:

- `Block` type
- `BlockHash` type
- `Cache` type

arguments of the macro and refer to all other traits through full qualified names.

Reply:
oh yeah?  At first I was going to add Block and Hash then I didn't.  Perhaps I was treating Rust macros too much like C macros?